File: test_shopee_all/4.r
library(dplyr)
library(lubridate)

customer <- read.csv("test_shopee_all/Customer.csv")


order <- read.csv("test_shopee_all/Orders.csv")
# order$Order.Date <- as.Date(order$Order.Date, format = '%m/%d/%Y')
order$Order.Date <- mdy(order$Order.Date)
order <- order %>%
    group_by(Customer.ID) %>%
    summarise(pembelian_awal = min(Order.Date)) %>%
    arrange(Customer.ID)


print(head(customer))
print(head(order))
cat("\n\n\n")

hasil <- merge(customer, order, by = "Customer.ID")
print(head(hasil))

hasil_akhir <- hasil %>%
    group_by(City, pembelian_awal) %>%
    summarize(jumlah = n()) %>%
    arrange(desc(jumlah))
print(head(hasil_akhir))
print(tail(hasil_akhir))

File: test_shopee_all/4.py
import pandas as pd

customer = pd.read_csv('test_shopee_all/Customer.csv')

order = pd.read_csv('test_shopee_all/Orders.csv')
order['Order Date'] = pd.to_datetime(order['Order Date'])
order = order.groupby('Customer ID')['Order Date'].min().reset_index(name = 'pembelian_pertama')

print(customer.head(5))
print(order.head(5))

hasil = pd.merge(customer, order, on = 'Customer ID',how='inner')
print(hasil.head())

hasil_akhir = hasil.groupby(['City', 'pembelian_pertama'])['Customer ID'].count().reset_index(name = 'jumlah').sort_values('jumlah', ascending=False).reset_index(drop=True)
print(hasil_akhir)